src/Hooks/FiltersBlock.md

```diff
@@ -0,0 +1,333 @@
+
+**Filter name:**
+`es_forms_block_field_style_options`
+
+**Available options:**
+ * input
+ * textarea
+ * checkboxes
+ * radios
+ * sender-email
+ * select
+ * file
+ * submit
+
+**Filter example:**
+```php
+// Set block field style options.
+add_filter('es_forms_block_field_style_options', [$this, 'getBlockFieldStyleOptions']);
+
+/**
+ * Set block field style options.
+ *
+ * @return array<string, mixed>
+ */
+public function getBlockFieldStyleOptions(): array
+{
+	return [
+		'input' => [
+			[
+				'label' => 'Default',
+				'value' => 'default'
+			],
+			[
+				'label' => 'Custom Style',
+				'value' => 'custom-style'
+			],
+		],
+		'select' => [
+			[
+				'label' => 'Default',
+				'value' => 'default'
+			],
+			[
+				'label' => 'Custom Style',
+				'value' => 'custom-style',
+				'useCustom' => false, // This key can be used only on select, file and textarea and it removes the custom JS library from the component.
+			],
+		]
+	];
+}
+```
+
+## Add to custom data block
+These filters will add the necessary data for the custom data block to work. Field data option selector will not be shown unless a filter is added. This option is showed in Block Editor.
+
+**Filter name:**
+`es_forms_block_custom_data_options`
+
+**Filter example options:**
+```php
+// Set block custom data options.
+add_filter('es_forms_block_custom_data_options', [$this, 'getBlockCustomDataOptions']);
+
+/**
+ * Set block custom data options.
+ *
+ * @return array<string, mixed>
+ */
+public function getBlockCustomDataOptions(): array
+{
+	return [
+		[
+			'label' => '',
+			'value' => ''
+		],
+		[
+			'label' => 'Blog posts',
+			'value' => 'blog-posts'
+		],
+		[
+			"label" => "Jobs",
+			"value" => "jobs"
+		],
+	];
+}
+```
+
+**Filter example option data:**
+```php
+// Set block custom data options data.
+add_filter('es_forms_block_custom_data_options_data', [$this, 'getBlockCustomDataOptionsData']);
+
+/**
+ * Set block custom data options data.
+ *
+ * @param string $type Type of option selected in the Block editor.
+ *
+ * @return array<string, mixed>
+ */
+public function getBlockCustomDataOptionsData(string $type): array
+{
+	switch ($type) {
+		case 'blog-posts':
+			return [
+				[
+					'label' => '',
+					'value' => ''
+				],
+				[
+					'label' => 'Post 1',
+					'value' => 'post1'
+				],
+				[
+					"label" => "Post 2",
+					"value" => "post2"
+				],
+			];
+		case 'jobs':
+			return [
+				[
+					'label' => '',
+					'value' => ''
+				],
+				[
+					'label' => 'Job 1',
+					'value' => 'job1'
+				],
+				[
+					"label" => "Job 2",
+					"value" => "job2"
+				],
+			];
+		default:
+			return [];
+	}
+}
+```
+
+## Override default submit button with your own component
+These filter will remove the default forms submit button component and use your callback. This will not apply to form settings pages.
```

Comment:
```suggestion
This filter will remove the default forms submit button component and use your callback. This will not apply to form settings pages.
```